### PR TITLE
Integrate new model

### PR DESCRIPTION
**Pull request by codymccoy**

### CODE REVIEW

File: ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs [hunk 1]
```diff
@@ -387,15 +382,22 @@ private void StageDamageSelectionChanged()
...
                 DamageCategories.AddRange(stageDamageCurves);
                 SelectedDamageCurve = DamageCategories[0];
             }
         }
 
+        //todo: check that these selected items aren't null?
+        private List<StageDamageCurve> GetStageDamageCurves()
+        {
+            AggregatedStageDamageElement elem = (AggregatedStageDamageElement)SelectedStageDamageElement.ChildElement;
+            List<StageDamageCurve> stageDamageCurves = elem.Curves.Where(curve => curve.ImpArea.ID == CurrentImpactArea.ID).ToList();
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs:397]: You don't need to keep around an output variable, just return elem.Curves.Where(...)

File: ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs [hunk 2]
```diff
@@ -435,15 +437,56 @@ public FdaValidationResult IsValid()
...
+            {
+                metrics.Results result = simulation.Compute(mrp, 1);
+                Console.WriteLine("Mean ead: " + result.ExpectedAnnualDamageResults.MeanEAD("InteriorStageDamage"));
+                double ead = result.ExpectedAnnualDamageResults.MeanEAD("InteriorStageDamage");
+                double total = result.ExpectedAnnualDamageResults.MeanEAD("Total");
+                int i = 0;
+            }
+            catch (Exception e)
+            {
+                MessageBox.Show(e.Message);
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs:472]: This strips the stack trace, we need to log that in some capacity, I'm not sure how though.  don't we use NLog?

File: ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs [hunk 3]
```diff
@@ -435,15 +437,56 @@ public FdaValidationResult IsValid()
...
+            }
         }
+
+        private List<UncertainPairedData> GetStageDamagesAsPairedData()
+        {
+            List<UncertainPairedData> stageDamages = new List<UncertainPairedData>();
+            List<StageDamageCurve> stageDamageCurves = GetStageDamageCurves();
+            foreach (StageDamageCurve curve in stageDamageCurves)
+            {
+                //todo: i don't like this.
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs:482]: Why don't you like this?  or is this part of the call we're in?

File: ViewModel/ImpactAreaScenario/IASElementSet.cs [hunk 4]
```diff
@@ -159,24 +156,22 @@ private ObservableCollection<ImpactAreaRowItem> GetStudyImpactAreaRowItems()
 
         private List<SpecificIASResultVM> GetResults()
         {
-            
             List<SpecificIASResultVM> results = new List<SpecificIASResultVM>();
-
             //this is kind of messy. Quite a bit of code to get the name of the impact area from the impact area id.
+            //todo: get a list of result objects
+            List<metrics.Results> iasResults = new List<metrics.Results>();
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/IASElementSet.cs:162]: Namespacing

File: ViewModel/ImpactAreaScenario/Results/DamageWithUncertaintyVM.cs [hunk 5]
```diff
@@ -16,8 +16,9 @@ public class DamageWithUncertaintyVM : BaseViewModel
 
         public List<EadRowItem> Rows { get; } = new List<EadRowItem>();
         public double Mean { get; set; }
-        public DamageWithUncertaintyVM()
+        public DamageWithUncertaintyVM(metrics.Results iasResult)
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Results/DamageWithUncertaintyVM.cs:19]: Namespacing

File: ViewModel/ImpactAreaScenario/Results/SpecificIASResultVM.cs [hunk 6]
```diff
@@ -20,7 +20,7 @@ public class SpecificIASResultVM : BaseViewModel
         private readonly List<string> _damageReports = new List<string>() { DAMAGE_WITH_UNCERTAINTY, DAMAGE_BY_DAMCAT };
 
         private readonly List<string> _performanceReports = new List<string>() { ANNUAL_EXC_PROB, LONG_TERM_RISK, ASSURANCE_OF_THRESHOLD };
-
+        private readonly metrics.Results _IASResult;
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Results/SpecificIASResultVM.cs:23]: Namespacing

File: ViewModel/ImpactAreaScenario/SpecificIAS.cs [hunk 7]
```diff
@@ -29,7 +37,7 @@ public class SpecificIAS
         /// These are the results after doing a compute. If a compute has not been
         /// done, then this will be null.
         /// </summary>
-        public IConditionLocationYearResult ComputeResults { get; set; }
+        public metrics.Results ComputeResults { get; set; }
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/SpecificIAS.cs:40]: Namespacing

File: ViewModel/ImpactAreaScenario/SpecificIAS.cs [hunk 8]
```diff
@@ -117,9 +125,36 @@ public SpecificIAS(XElement iasElem)
...
+                metrics.Results result = simulation.Compute(mrp, 1);
+                Console.WriteLine("Mean ead: " + result.ExpectedAnnualDamageResults.MeanEAD("InteriorStageDamage"));
+                double ead = result.ExpectedAnnualDamageResults.MeanEAD("InteriorStageDamage");
+                double total = result.ExpectedAnnualDamageResults.MeanEAD("Total");
+                ComputeResults = result;
+                return result;
+            }
+            catch (Exception e)
+            {
+                MessageBox.Show(e.Message);
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/SpecificIAS.cs:155]: Losing the stack trace here, we should probably log it somehow.

**Review by RyanM-RMA — CHANGES_REQUESTED** [review]
Looking good for the most part.  Namespace changes requested, and additional logging of two exceptions.

File: ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs (hunk 2 above)
Comment (codymccoy) [reply ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs:472]: We don't really log anything at this point. Just leave it?

File: ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs (hunk 3 above)
Comment (codymccoy) [reply ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs:482]: I changed the todo to be more descriptive of what i don't like. 
                //todo: i don't like that i have to take the icoordinates funtion and turn it into an ifda function to turn it into a paired data object.

File: ViewModel/ImpactAreaScenario/IASElementSet.cs (hunk 4 above)
Comment (codymccoy) [reply ViewModel/ImpactAreaScenario/IASElementSet.cs:162]: I tried to get rid of "metrics" but "Results" is also a namespace and VS can't seem to figure it out without the full path.

**Comment by rnugent3** [comment]
Yikes that is messy. I’ll add this to my list.

**Comment by RyanM-RMA** [comment]
> Yikes that is messy. I’ll add this to my list.
> […](#)
> On Dec 16, 2021, at 4:56 PM, codymccoy ***@***.***> wrote: ﻿ @codymccoy commented on this pull request. In ViewModel/ImpactAreaScenario/IASElementSet.cs: > //this is kind of messy. Quite a bit of code to get the name of the impact area from the impact area id. + //todo: get a list of result objects + List<metrics.Results> iasResults = new List<metrics.Results>(); I tried to get rid of "metrics" but "Results" is also a namespace and VS can't seem to figure it out without the full path. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.

@rnugent3 does that mean you're creating an issue for it, and you'll be updating this code once it's ready? I'd like to close this PR unless you want hold off because of this code.

**Comment by rnugent3** [comment]
I’m willing to accept this particular messiness for now. After all, it is my mess. I have an issue open already.  Thanks, Cody.

**Review by RyanM-RMA — APPROVED** [review]

**Pull request merged**